exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/BaseParquetMetadataProvider.java

```diff
@@ -87,6 +88,7 @@
   private TableMetadata tableMetadata;
   private List<PartitionMetadata> partitions;
   private Map<Path, FileMetadata> files;
+  private NonInterestingColumnsMetadata nonInterestingColumnsMetadata;
```

Comment:
For older metadata versions (V3.3 and prior) will this be initialized to null  ?  Have you tried with prior versions ?

Reply:
Column statistics will be an empty map in case of older metadata versions. This is handled by checking if parquetTableMetadata is an instance of Metadata_V4.ParquetTableMetadata_v4.

Reply:
Some of the unit tests actually use the metadata cache files from older versions.